src/sinol_make/commands/export/__init__.py

```diff
@@ -0,0 +1,140 @@
+            if os.path.exists(file_path):
+                if os.path.isdir(file_path):
+                    shutil.copytree(file_path, os.path.join(target_dir, file))
+                else:
+                    shutil.copy(file_path, target_dir)
+
+        print('Generating tests...')
+        generated_tests = self.get_generated_tests()
+        print('Copying handwritten and example tests...')
+        for ext in ['in', 'out']:
+            os.mkdir(os.path.join(target_dir, ext))
+            for test in glob.glob(os.path.join(os.getcwd(), ext, f'{self.task_id}*.{ext}')) + \
+                    glob.glob(os.path.join(os.getcwd(), ext, f'{self.task_id}0*.{ext}')):
+                if package_util.extract_test_id(test) not in generated_tests:
+                    shutil.copy(test, os.path.join(target_dir, ext))
+
+    def create_makefile_in(self, target_dir: str, config: dict):
+        """
+        Creates required `makefile.in` file.
+        :param target_dir: Directory to create files in.
+        :param config: Config dictionary.
+        """
+        with open(os.path.join(target_dir, 'makefile.in'), 'w') as f:
+            cxx_flags = '-std=c++17'
+            c_flags = '-std=c17'
+            if 'extra_compilation_args' in config:
+                if 'cpp' in config['extra_compilation_args']:
+                    cxx_flags += ' ' + ' '.join(config['extra_compilation_args']['cpp'])
+                if 'c' in config['extra_compilation_args']:
+                    c_flags += ' ' + ' '.join(config['extra_compilation_args']['c'])
+
+            f.write(f'MODE = wer\n'
+                    f'ID = {self.task_id}\n'
+                    f'SIG = sinolmake\n'
+                    f'\n'
+                    f'TIMELIMIT = {config["time_limit"]}\n'
+                    f'SLOW_TIMELIMIT = {4 * config["time_limit"]}\n'
+                    f'MEMLIMIT = {config["memory_limit"]}\n'
+                    f'\n'
+                    f'OI_TIME = oiejq\n'
+                    f'\n'
+                    f'CXXFLAGS += {cxx_flags}\n'
+                    f'CFLAGS += {c_flags}\n')
+
+    def compress(self, tmpdir, target_dir):
+        """
+        Compresses target directory to archive.
+        :param tmpdir: Temporary directory path.
+        :param target_dir: Target directory path.
+        :return: Path to archive.
+        """
+        archive = os.path.join(tmpdir, f'{self.task_id}.tgz')
+        with tarfile.open(archive, "w:gz") as tar:
+            tar.add(target_dir, arcname=os.path.basename(target_dir))
+        return archive
+
+    def run(self, args: argparse.Namespace):
+        util.exit_if_not_package()
+
+        self.args = args
+        export_dir = args.output or 'export'
+        self.task_id = package_util.get_task_id()
+        self.export_dir = os.path.abspath(export_dir)
+        if not os.path.exists(self.export_dir):
+            os.makedirs(self.export_dir)
+
+        with tempfile.TemporaryDirectory() as tmpdir:
```

Comment:
Did we settle the "cache vs tempdir" issue? I would prefer to be consistent in the use of those two directories.

Reply:
Yes, I changed it to use `cache` dir for exporting